fe/fe-core/src/main/java/com/starrocks/alter/dynamictablet/DynamicTabletJob.java

```diff
@@ -211,10 +228,370 @@ public void run() {
+            return false;
+        }
+    }
+
+    private void replacePhysicalPartitions() {
+        try (LockedObject<OlapTable> lockedTable = getLockedTable(LockType.WRITE)) {
+            OlapTable olapTable = lockedTable.get();
+            for (var physicalPartitionEntry : physicalPartitionContexts.entrySet()) {
+                PhysicalPartition physicalPartition = olapTable.getPhysicalPartition(physicalPartitionEntry.getKey());
+                if (physicalPartition == null) {
+                    continue;
+                }
+                PhysicalPartitionContext physicalPartitionContext = physicalPartitionEntry.getValue();
+
+                long commitVersion = physicalPartitionContext.getCommitVersion();
+                Preconditions.checkState(commitVersion == physicalPartition.getVisibleVersion() + 1,
+                        "commit version: " + commitVersion + ", visible version: "
+                                + physicalPartition.getVisibleVersion());
+                physicalPartition.setVisibleVersion(commitVersion, finishedTimeMs);
+
+                PhysicalPartition newPhysicalPartition = new PhysicalPartition(
+                        physicalPartitionContext.getNewPhysicalPartitionId(),
+                        physicalPartition);
+
+                Map<Long, MaterializedIndexContext> indexContexts = physicalPartitionContext.getIndexContexts();
+                for (MaterializedIndexContext indexContext : indexContexts.values()) {
+                    MaterializedIndex newIndex = indexContext.getMaterializedIndex();
+                    if (newIndex.getId() == physicalPartition.getBaseIndex().getId()) {
+                        newPhysicalPartition.setBaseIndex(newIndex);
+                    } else {
+                        newPhysicalPartition.createRollupIndex(newIndex);
+                    }
+                }
+
+                olapTable.replacePhysicalPartition(physicalPartition.getId(), newPhysicalPartition, true);
+            }
+        }
+    }
+
+    private void removePhysicalPartitions() {
+        TabletInvertedIndex invertedIndex = GlobalStateMgr.getCurrentState().getTabletInvertedIndex();
+        try (LockedObject<OlapTable> lockedTable = getLockedTable(LockType.WRITE)) {
+            OlapTable olapTable = lockedTable.get();
+            for (long physicalPartitionId : physicalPartitionContexts.keySet()) {
+                PhysicalPartition physicalPartition = olapTable.removePhysicalPartition(physicalPartitionId);
+                if (physicalPartition == null) {
+                    continue;
+                }
+
+                for (MaterializedIndex index : physicalPartition.getMaterializedIndices(IndexExtState.ALL)) {
+                    for (Tablet tablet : index.getTablets()) {
+                        invertedIndex.deleteTablet(tablet.getId());
+                    }
+                }
+            }
+        }
+    }
+
+    private void addTabletsToInvertedIndex() {
```

Comment:
Shall we need to create physical tablet for the split/merge? It seems that it is just some metadata operation.